src/rust/engine/src/externs/interface.rs

```diff
@@ -685,12 +686,20 @@ async fn workunit_to_py_value(
       ),
     ];
 
-    if let Some(parent_id) = workunit.parent_id {
-      dict_entries.push((
-        externs::store_utf8(py, "parent_id"),
-        externs::store_utf8(py, &format!("{}", parent_id)),
-      ));
+    let parent_ids = workunit
+      .parent_ids
+      .into_iter()
+      .map(|parent_id| externs::store_utf8(py, &parent_id.to_string()))
+      .collect::<Vec<_>>();
+
+    if has_parent_ids {
+      // TODO: Remove the single-valued `parent_id` field around version 2.16.0.dev0.
+      dict_entries.push((externs::store_utf8(py, "parent_id"), parent_ids[0].clone()));
     }
+    dict_entries.push((
+      externs::store_utf8(py, "parent_ids"),
```

Comment:
Ooooh. Fun!
Is there any order to the parent IDs? E.g. does the first ID have any semantic meaning?

Reply:
There will be implicitly, since I believe that the order of addition of parents will be preserved, but I hadn't thought about trying to make any guarantees there. What did you have in mind?

Reply:
Just thinking about how previously the parent was the first parent to trigger this rule, and wondering if that was still the case with the first parent ID. It's likely not relevant enough to make a strong guarantee.